docker/commands/gui.py

```diff
@@ -309,28 +309,34 @@ async def show_promote_confirmation(self, widget: Widget, key: blessed.keyboard.
         task = data["task"]
 
         recent_deployment = await self._gh.get_recent_deployment(next_env)
-        git_log_lines = get_git_log(recent_deployment["ref"], ref) if recent_deployment else None
-        if git_log_lines is not None:
-            git_log = (
-                "\n".join(git_log_lines)
-                if git_log_lines
-                else (colorama.Fore.CYAN + "No additional commits" + colorama.Fore.RESET)
-            )
+
+        if recent_deployment:
+            commits, completed = await self._gh.get_commits_until(ref, recent_deployment["ref"])
+
+            def commit_to_oneline(commit):
+                committer = commit["commit"]["committer"]
+                author = committer["name"]
+                committed = localize_date(committer["date"])
+                message = commit["commit"]["message"].splitlines()[0]
+                sha = short_sha(commit["sha"])
+                return f"[{sha}  {committed}  {author}]  {message}"
+
+            if not commits:
+                git_log = colorama.Fore.CYAN + "No additional commits" + colorama.Fore.RESET
+            else:
+                git_log_lines = list(reversed(list(map(commit_to_oneline, commits))))
```

Comment:
```suggestion
                git_log_lines = list(map(commit_to_oneline, commits))[::-1]
```
?

Reply:
I'd like to see the list of changes from oldes to newest, so I can read the message from top to bottom without changing reading direction for the commit list just to read the list of commits ordered by date.

Reply:
The code is equivalent isn't it, it's just that IMO doesn't make sense to call `reversed` to produce a generator, which will be then instantly evaluated in a `list`, but rather reverse the list coming out of `map` with `[::-1]` directly producing a new list.

Reply:
Agreed, somehow I missed that `[::-1]`.